### PR TITLE
Allow disabling the transport by plane even from default origin

### PR DESCRIPTION
**Pull request by magopian**
[user story](https://www.notion.so/fd0068a7db8348c584cd53a3eea399f2?v=90c1a9d44ff845c090b6f491065ba2ef&p=2786364aac184ee88a14b480045ebe61&pm=s)

Follows on #220 by allowing the disabling of the transport by plane for category 4 ingredients (default origin by plane) even from their default origin.

### CODE REVIEW

**Review by ccomb — APPROVED** [review]
Since a (Maybe Bool) is a set with 3 states, it might be made more explicit by defining a 3-state sum type for the byPlane with 3 constructors ?

**Comment by magopian** [comment]
I've pushed a commit to use a union type, which made a significant change, would you mind reviewing again @ccomb and see if that's what you had in mind?

File: src/Server/Query.elm [hunk 1]
```diff
@@ -230,17 +233,17 @@ validateBool str =
...
         "byPlane" ->
-            Ok (Just True)
+            Ok Ingredient.ByPlane
 
         "noPlane" ->
-            Ok (Just False)
+            Ok Ingredient.NoPlane
 
         _ ->
             Err "La valeur ne peut être que parmis les choix suivants: 'default', 'byPlane', 'noPlane'."
```
Comment (ccomb) [inline src/Server/Query.elm:249]: parmis→parmi

**Review by ccomb — APPROVED** [review]
:+1: It's easier to understand with this explicit type !

**Pull request merged**